tests/test_handlers.py

```diff
@@ -126,25 +126,77 @@ def test_vt_http_error(self, mock_requests_get, domain_handler, capsys):
         mock_requests_get.return_value = mock_resp
 
         # Thorough test of first _fetch_* method
-        with pytest.raises(SystemExit) as e:
+        with pytest.raises(HandlerException) as e:
             handler._fetch_vt_domain()
 
+        assert (
+            e.value.args[0]
+            == "Error fetching data: 401 Client Error: None for url: None"
+        )
+        assert e.type is HandlerException  # ruff E721
+
+        # Extra: just make sure program exits correctly
+        with pytest.raises(HandlerException) as e:
+            handler._fetch_vt_resolutions()
+
+    @patch.object(requests.Session, "get")
+    def test_vt_http_error2(self, mock_requests_get, domain_handler, capsys):
```

Comment:
Amazing! This is good, just need to move this to [test_cli.py](https://github.com/pirxthepilot/wtfis/blob/main/tests/test_cli.py). Please create a new class, e.g. `TestFetchData`.